bedms/train.py

```diff
@@ -0,0 +1,242 @@
+            y_test_list,
+            x_values_val_list,
+            x_headers_val_list,
+            y_val_list,
+            num_cluster,
+            vectorizer_pth,
+            label_encoder_pth,
+            sentence_transformer_model,
+        )
+        logger.info("Encoding Done.")
+
+        batch_size = self.config["training"]["batch_size"]
+        self.train_loader = data_loader(train_encoded_data, batch_size)
+        self.test_loader = data_loader(test_encoded_data, batch_size)
+        self.val_loader = data_loader(val_encoded_data, batch_size)
+
+        logger.info("Loading Done.")
+
+    def training(self):
+        """
+        Trains the model.
+        """
+        input_size_values = len(self.vectorizer.vocabulary_)
+        input_size_values_embeddings = EMBEDDING_SIZE
+        input_size_headers = EMBEDDING_SIZE
+        hidden_size = self.config["model"]["hidden_size"]
+        self.output_size = len(self.label_encoder.classes_)  # Number of classes
+        dropout_prob = self.config["model"]["dropout_prob"]
+
+        self.model = BoWSTModel(
+            input_size_values,
+            input_size_values_embeddings,
+            input_size_headers,
+            hidden_size,
+            self.output_size,
+            dropout_prob,
+        )
+
+        learning_rate = self.config["training"]["learning_rate"]
+        self.criterion = nn.CrossEntropyLoss()
+        l2_reg_lambda = self.config["training"]["l2_regularization"]
+        optimizer = optim.Adam(
+            self.model.parameters(), lr=learning_rate, weight_decay=l2_reg_lambda
+        )
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        # Training the model
+        num_epochs = self.config["training"]["num_epochs"]
+
+        model_pth = self.config["training"]["model_pth"]
+        bow_drops = self.config["training"]["bow_drops"]
+
+        (
+            self.train_accuracies,
+            self.val_accuracies,
+            self.train_losses,
+            self.val_losses,
+            self.fpr,
+            self.tpr,
+            self.roc_auc,
+        ) = train_model(
+            self.model,
+            self.train_loader,
+            self.val_loader,
+            self.criterion,
+            optimizer,
+            device,
+            num_epochs,
+            self.output_size,
+            model_pth,
+            bow_drops,
+        )
+
+        logger.info("Training Done.")
+
+    def testing(self):
+        """
+        Model testing.
+        """
+        self.all_preds, self.all_labels = model_testing(
+            self.model, self.test_loader, self.criterion
+        )
+        precision = precision_score(self.all_labels, self.all_preds, average="macro")
+        recall = recall_score(self.all_labels, self.all_preds, average="macro")
+        f1 = f1_score(self.all_labels, self.all_preds, average="macro")
+        logger.info(f"Precision:{precision}, Recall: {recall}, F1 Score: {f1}")
```

Comment:
I would return these values. the user probably wants them for themselves anyways

Reply:
Like:
```python
return {
  'precision': precision,
  'recall': recall,
  'f1': f1
}
```